### PR TITLE
[Eager] pylayer detach output tensor if it is equal with input

### PR DESCRIPTION
**Pull request by wanghuancoder**
### PR types

Others 
### PR changes

Others 
### Describe

对于输入也是输出的场景，PyLayer需要detach一下，否则构建的反向图是错的。

### CODE REVIEW

File: paddle/fluid/pybind/eager_py_layer.cc [hunk 1]
```diff
@@ -108,6 +108,20 @@ PyObject* pylayer_method_name(PyObject* self, PyObject* noargs) {
   EAGER_CATCH_AND_THROW_RETURN_NULL
 }
 
+PyObject* tensor_view(paddle::experimental::Tensor* tensor) {
```
Comment (JiabinYang) [inline paddle/fluid/pybind/eager_py_layer.cc]: Better to change another name tensor view is not fully supported with this method

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by chenwhql — DISMISSED** [review]

**Review by momozi1996 — COMMENTED** [review]
LGTM

File: paddle/fluid/pybind/eager_py_layer.cc (hunk 1 above)
Comment (wanghuancoder) [reply paddle/fluid/pybind/eager_py_layer.cc]: done

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Review by momozi1996 — APPROVED** [review]
LGTM

**Pull request merged**